src/indexer1.rs

```diff
@@ -144,8 +147,12 @@ impl<S: LogStorage, P: Processor<S::Transaction>> Indexer<S, P> {
     }
 
     async fn handle_tick(&mut self) -> anyhow::Result<()> {
-        let to_block = self.provider.get_block_number().await?;
+        let mut to_block = self.provider.get_block_number().await?;
         let from_block = self.last_observed_block + 1;
+        if to_block-from_block > self.block_range_limit {
```

Comment:
This snippet matched the optional limit and also looks more short to me:
```rust
let latest_block = self.provider.get_block_number().await?
let to_block = self.block_range_limit
             .map(| block_range_limit | 
                        latest_block.min(from_block + block_range_limit)
             ).unwrap_or(latest_block);
```
Also, pleas make a standalone commit later, that will ensure cargo format worked, it seems that ```to_block-from_block``` wasn't formatted.

Reply:
Done